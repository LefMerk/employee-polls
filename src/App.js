import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <div className='flex flex-col'>
        <h1>Employee Polls</h1>
        <div>1</div>
        <div>2</div>
      </div>
    </div>
  );
}

export default App;
