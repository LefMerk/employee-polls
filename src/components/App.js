import React, { useEffect } from 'react';
import { useSelector, useDispatch } from "react-redux";
import { Routes, Route } from 'react-router';
import ProtectedRoute from './ProtectedRoute';
import NavBar from "./NavBar";
import NotFound from "./NotFound";
import Login from "./Login";
import Dashboard from "./Dashboard";
import Leaderboard from "./Leaderboard";
import NewPoll from "./NewPoll";
import Poll from "./Poll";
import { handleInitialData } from '../actions/shared';

export default function App() {
  const userLoggedIn = useSelector(state => state.authedUser);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(handleInitialData());
  });

  return (
    <div className="App">
      {userLoggedIn && userLoggedIn !== 'Not authorised'
        ? <>
            <NavBar />
            <Routes>
              <Route path="*" element={<NotFound />}></Route>
              <Route
                exact 
                path="/" 
                element={
                  <ProtectedRoute>
                    <Dashboard />
                  </ProtectedRoute>
                }
              />
              <Route
                exact 
                path="/leaderboard" 
                element={
                  <ProtectedRoute>
                    <Leaderboard />
                  </ProtectedRoute>
                }
              />
              <Route 
                exact 
                path="/add" 
                element={
                  <ProtectedRoute>
                    <NewPoll />
                  </ProtectedRoute>
                } 
              />
              <Route 
                exact 
                path="/question/:id" 
                element={
                  <ProtectedRoute>
                    <Poll />
                  </ProtectedRoute>
                }
              />
            </Routes>
          </>
        : <Login />
      }
    </div>
  );
}