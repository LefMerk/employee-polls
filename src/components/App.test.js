import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import { store } from '../store';
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";

jest.mock('../assets/login_poll.png', () => 'login_poll.png');

describe('App', () => {
    it('will render without crashing', () => {
        const wrapper = render(
            <Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('will render the login page', () => {
        const wrapper = render(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/login']}>
                    <App />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.getByTestId('login-btn')).toBeInTheDocument();
    });
});