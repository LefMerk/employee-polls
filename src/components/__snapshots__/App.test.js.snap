// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will render without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="flex flex-col items-center gap-8"
        >
          <h1
            class="text-3xl mt-8 font-semibold"
          >
            Employee Polls
          </h1>
          <img
            alt="login_image"
            class="max-w-sm"
            src="login_poll.png"
          />
          <h2
            class="text-2xl font-semibold"
          >
            Log In
          </h2>
          <form
            class="flex flex-col items-center w-60"
          >
            <label
              class="font-semibold mb-1"
              for="user"
            >
              User
            </label>
            <input
              autocomplete="username"
              class="mb-4 p-2 block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-sky-700 sm:text-sm sm:leading-6"
              data-testid="user"
              id="user"
              name="user"
              required=""
              type="text"
              value=""
            />
            <label
              class="font-semibold mb-1"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="mb-4 p-2 block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-sky-700 sm:text-sm sm:leading-6"
              data-testid="password"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <button
              class="rounded-md bg-sky-700 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-cyan-600 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-sky-700"
              data-testid="login-btn"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="flex flex-col items-center gap-8"
      >
        <h1
          class="text-3xl mt-8 font-semibold"
        >
          Employee Polls
        </h1>
        <img
          alt="login_image"
          class="max-w-sm"
          src="login_poll.png"
        />
        <h2
          class="text-2xl font-semibold"
        >
          Log In
        </h2>
        <form
          class="flex flex-col items-center w-60"
        >
          <label
            class="font-semibold mb-1"
            for="user"
          >
            User
          </label>
          <input
            autocomplete="username"
            class="mb-4 p-2 block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-sky-700 sm:text-sm sm:leading-6"
            data-testid="user"
            id="user"
            name="user"
            required=""
            type="text"
            value=""
          />
          <label
            class="font-semibold mb-1"
            for="password"
          >
            Password
          </label>
          <input
            autocomplete="current-password"
            class="mb-4 p-2 block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-sky-700 sm:text-sm sm:leading-6"
            data-testid="password"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <button
            class="rounded-md bg-sky-700 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-cyan-600 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-sky-700"
            data-testid="login-btn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
